File: fontend/website/src/types/attributeOption.ts
import type { AttributeDefinition } from "./attributeDefinition";

export interface AttributeOption {
  id: number;
  attributeDefinitionId: number;
  attributeDefinition?: AttributeDefinition
  valueKey: string;
  label: string;
  displayOrder: number;
}
export interface CreateAttributeOptionDTO {
  attributeDefinitionId?: number;
  attributeDefinition?: AttributeDefinition
  valueKey?: string;
  label?: string;
  displayOrder?: number;
}

export interface UpdateAttributeOptionDTO {
  attributeDefinitionId?: number;
  attributeDefinition?: AttributeDefinition
  valueKey?: string;
  label?: string;
  displayOrder?: number;
}

export interface OptionFilterttributeOption {
  page?: number,
  size?: number,
  attributeDefinitionId?: number,
  keySearch?: string,
}

export interface Response {
  total: number;
  page: number;
  size: number;
  items: AttributeOption[]
}

File: backend/DienMayLongQuyen.Api/Models/AttributeOption.cs
namespace DienMayLongQuyen.Api.Models
{
    public class AttributeOption
    {
        public int Id { get; set; }

        public int AttributeDefinitionId { get; set; }
        public AttributeDefinition AttributeDefinition { get; set; }

        // Giá trị logic (VD: "2", "4", "CuaNgang")
        public string ValueKey { get; set; }

        // Nhãn hiển thị cho UI (VD: "2 cánh", "4 cánh", "Cửa ngang")
        public string Label { get; set; }

        // Thứ tự hiển thị trong dropdown/filter UI
        public int DisplayOrder { get; set; }

        public ICollection<ProductAttributeOption> ProductAttributeOptions { get; set; }
    }
}